godotproject/Global/ResourceQueue.gd

```diff
@@ -0,0 +1,145 @@
+var thread
+var mutex
+var sem
```

Comment:
What are these semaphores used for?

Reply:
I think they're used to manage sharing resources between the foreground "normal game" thread and the background loading thread (a guess since I didn't write the code :P)